### PR TITLE
Architecture overview diagram

### LINKED ISSUE #9: Diagrams should be as editable as markdown code
### Is your feature request related to a problem? Please describe

When we want to describe a system, make a flow chart, or an activity diagram there should be a descriptive language that is used and can be edited without using another tool.

### Describe the solution you'd like

Inline detection via a GithubAction that converts plantuml into images would be great.

### PR DESCRIPTION
**Pull request by peternied**
Added a workflow that will automatically generate puml files into svg and commit them into the repository.

Signed-off-by: Peter Nied <petern@amazon.com>

### Issues
- Resolves #9 

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
See a preview with the 'rich diff' button
image

File: README.md [hunk 1]
```diff
@@ -12,6 +12,14 @@
 
 OpenSearch Continous Integration is an open source CI system for OpenSearch and its plugins.
 
+### Architecture Overview
+
+![Architecture Diagram](.\img\opensearch-ci-overview.png)
```
Comment (dblock) [inline README.md:17]: This doesn't seem to render, need forward slashes?

File: README.md [hunk 2]
```diff
@@ -12,6 +12,14 @@
...
+
+[Source](https://gist.github.com/peternied/55db1ce5277a61923519364dd6271ce5)
```
Comment (dblock) [inline README.md:19]: Migrate into this repo.

**Review by dblock — CHANGES_REQUESTED** [review]

**Comment by dblock** [comment]
I would think the generation job would PR the outputs when they change or otherwise maybe GHA is not the right approach here and you want some kind of precommit hook to update the SVGs?

**Review by dblock — APPROVED** [review]

**Pull request merged**